chemprop/uncertainty/evaluator.py

```diff
@@ -33,22 +35,34 @@ def evaluate(self, preds: Tensor, uncs: Tensor, targets: Tensor, mask: Tensor) -
 @UncertaintyEvaluatorRegistry.register("nll-regression")
 class NLLRegressionEvaluator(UncertaintyEvaluator):
     def evaluate(self, preds: Tensor, uncs: Tensor, targets: Tensor, mask: Tensor) -> Tensor:
-        ...
-        return
+        masked_preds = preds * mask
+        masked_targets = targets * mask
+        masked_uncs = uncs * mask
+        nlls = (2 * torch.pi * masked_uncs).log() / 2 + (masked_preds - masked_targets) ** 2 / (
+            2 * masked_uncs
+        )
+        return nlls.mean(dim=0)
 
 
 @UncertaintyEvaluatorRegistry.register("nll-classification")
 class NLLClassEvaluator(UncertaintyEvaluator):
     def evaluate(self, preds: Tensor, uncs: Tensor, targets: Tensor, mask: Tensor) -> Tensor:
-        ...
-        return
+        masked_targets = targets * mask
+        masked_uncs = uncs * mask
+        likelihoods = masked_uncs * masked_targets + (1 - masked_uncs) * (1 - masked_targets)
+        nlls = -1 * likelihoods.log()
+        return nlls.mean(dim=0)
 
 
 @UncertaintyEvaluatorRegistry.register("nll-multiclass")
 class NLLMultiEvaluator(UncertaintyEvaluator):
     def evaluate(self, preds: Tensor, uncs: Tensor, targets: Tensor, mask: Tensor) -> Tensor:
-        ...
-        return
+        masked_targets = targets * mask
+        masked_uncs = uncs * mask.unsqueeze(dim=1)
```

Comment:
Is the shape of `masked_uncs` `n x t x c`?

Reply:
I just checked, and yes, `masked_uncs` should be of shape `n x t x c`.